src/main/java/de/unijena/cheminf/mortar/model/fragmentation/algorithm/AlkylStructureFragmenter.java

```diff
@@ -0,0 +1,1446 @@
+        this.settingNameTooltipTextMap.put(this.altHandlingSingleTertQuatCarbonsSetting.getName(),
+                Message.get("AlkylStructureFragmenter.altHandlingSingleTertQuatCarbonsSetting.tooltip"));
+        this.settingNameDisplayNameMap.put(this.altHandlingSingleTertQuatCarbonsSetting.getName(),
+                Message.get("AlkylStructureFragmenter.altHandlingSingleTertQuatCarbonsSetting.displayName"));
+        this.mcbSingleRingDetectionSetting = new SimpleBooleanProperty(this, "Single ring detection setting",
+                AlkylStructureFragmenter.MCB_SINGLE_RING_DETECTION_SETTING_DEFAULT);
+        this.settingNameTooltipTextMap.put(this.mcbSingleRingDetectionSetting.getName(),
+                Message.get("AlkylStructureFragmenter.mcbSingleRingDetectionSetting.tooltip"));
+        this.settingNameDisplayNameMap.put(this.mcbSingleRingDetectionSetting.getName(),
+                Message.get("AlkylStructureFragmenter.mcbSingleRingDetectionSetting.displayName"));
+        this.separateTertQuatCarbonFromRingSetting = new SimpleBooleanProperty(this, "Separate tertiary and " +
+                "quaternary carbon atoms from ring structures setting", AlkylStructureFragmenter.SEPARATE_TERT_QUAT_CARBON_FROM_RING_SETTING_DEFAULT);
+        this.settingNameTooltipTextMap.put(this.separateTertQuatCarbonFromRingSetting.getName(),
+                Message.get("AlkylStructureFragmenter.separateTertQuatCarbonFromRingSetting.tooltip"));
+        this.settingNameDisplayNameMap.put(this.separateTertQuatCarbonFromRingSetting.getName(),
+                Message.get("AlkylStructureFragmenter.separateTertQuatCarbonFromRingSetting.displayName"));
+        this.settings = new ArrayList<>(7);
+        this.settings.add(this.fragmentSaturationSetting);
+        this.settings.add(this.keepNonFragmentableMoleculesSetting);
+        this.settings.add(this.fragmentSideChainsSetting);
+        this.settings.add(this.maxChainLengthSetting);
+        this.settings.add(this.altHandlingSingleTertQuatCarbonsSetting);
+        this.settings.add(this.mcbSingleRingDetectionSetting);
+        this.settings.add(this.separateTertQuatCarbonFromRingSetting);
+        //set chemObjectBuilderInstance
+        this.setChemObjectBuilderInstance();
+    }
+    //</editor-fold>
+    /**
+     * Inner class for internal data transfer of atoms and bonds in arrays.
+     */
+    class MolecularArrays {
+                private IAtom[] atomArray;
+                private IBond[] bondArray;
+                protected MolecularArrays(IAtomContainer aMolecule) {
+                    this.atomArray = new IAtom[aMolecule.getAtomCount()];
+                    this.bondArray = new IBond[aMolecule.getBondCount()];
+                }
+                protected MolecularArrays() {
+                    this.atomArray = new IAtom[1];
+                    this.bondArray = new IBond[1];
+                }
```

Comment:
Remove this constructor, it is unused and not good to initialise the arrays with length 1.